src/main/java/org/venice/beachfront/bfapi/config/OAuth2Config.java

```diff
@@ -0,0 +1,166 @@
+
+    @Autowired
+    private UserProfileService userProfileService;
+
+    @Bean
+    public OAuth2RestTemplate oauth2RestTemplate() {
+    	final OAuth2RestTemplate oauth2RestTemplate = new OAuth2RestTemplate(resourceDetails, oAuth2ClientContext);
+        oauth2RestTemplate.setAccessTokenProvider(accessTokenProviderChain());
+
+        return oauth2RestTemplate;
+    }
+
+    @Bean
+    public AccessTokenProvider accessTokenProviderChain() {
+        return new AccessTokenProviderChain(Arrays.<AccessTokenProvider> asList(
+            new OpenIDTokenProvider(),
+            new AuthorizationCodeAccessTokenProvider(),
+            new ImplicitAccessTokenProvider(),
+            new ResourceOwnerPasswordAccessTokenProvider(),
+            new ClientCredentialsAccessTokenProvider()));
+    }
+
+    @RequestMapping("/oauth/start")
+    public String authCode(final Model model, final HttpServletRequest request) throws Exception {
+
+    	if (ssoServiceUrl.equals("placeholder")) {
+            model.addAttribute("header", "Warning: You need to bind to the SSO service.");
+            model.addAttribute("warning", "Please bind your app to restore regular functionality");
+            return "configure_warning";
+        }
+
+        // Get information for the User from the /userinfo endpoint available on the SSO Tile
+        final Map<?,?> userInfoResponse = oauth2RestTemplate().getForObject("{ssoServiceUrl}/userinfo", Map.class, ssoServiceUrl);
+        model.addAttribute("ssoServiceUrl",ssoServiceUrl);
+        model.addAttribute("response", toPrettyJsonString(userInfoResponse));
+
+        // Get Access Token
+        final OAuth2AccessToken accessToken = oauth2RestTemplate().getOAuth2ClientContext().getAccessToken();
+        if (accessToken != null) {
+            model.addAttribute("access_token", toPrettyJsonString(parseToken(accessToken.getValue())));
+            model.addAttribute("id_token", toPrettyJsonString(parseToken((String) accessToken.getAdditionalInformation().get("id_token"))));
+        }
+
+        // Confirm User Profile exists; if not, create
+        final String userId = (String) userInfoResponse.get("user_id");
+        final String userName = (String) userInfoResponse.get("user_name");
+        getOrCreateUser(userId, userName);
+
+        return "authorization_code";
+    }
+
+    @RequestMapping(value="/logout", method = RequestMethod.GET)
+    public String logout(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+
+    	final Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth != null){
+            new SecurityContextLogoutHandler().logout(request, response, auth);
+        }
+        final URL url = new URL(request.getRequestURL().toString());
+        final String urlStr = url.getProtocol() + "://" + url.getAuthority();
+
+        return "redirect:" + ssoServiceUrl + "/logout.do?redirect=" + urlStr + "&clientId=" + clientId;
```

Comment:
Is this logout redirect url part of the configuration (whitelist) in PCF?
I finally got logout working in INT and i'm wondering if it's the same kind of deal